File: livingrimoire start here/LivinGrimoire java/older livingrimoire version/old java_core/ChobitsLight.java
package LG_Core;

import java.util.ArrayList;
import java.util.Hashtable;

public class ChobitsLight implements thinkable {
	protected String emot = ""; // emotion
	protected ArrayList<AbsCmdReq> dClassesLv1;
	// algorithms fusion (polymarization)
	protected Hashtable<String, Integer> AlgDurations;
	protected Fusion fusion;
	// region essential DClasses
	protected Permission permission;
	protected DPermitter dPermitter;
	// endregion
	protected Neuron noiron;
	// sleep vars :
	protected Person activePerson = new Person();
	// added :
	protected Kokoro kokoro; // soul
	protected Person master = new Person();
	protected String lastOutput = "";
	// standBy phase 260320
	protected ZeroTimeGate timeGate = new ZeroTimeGate();

	public ChobitsLight(PersonalityLight personality) {
		super();
		this.AlgDurations = personality.getAlgDurations();
		this.fusion = personality.getFusion();
		this.kokoro = personality.getKokoro();
		noiron = new Neuron();
		dClassesLv1 = personality.getdClassesLv1();
	}

	public void loadPersonality(PersonalityLight personality) {
		this.AlgDurations = personality.getAlgDurations();
		this.fusion = personality.getFusion();
		this.kokoro = personality.getKokoro();
		noiron = new Neuron();
		dClassesLv1 = personality.getdClassesLv1();
	}

	public String doIt(String ear, String skin, String eye) {
		ear = translateIn(ear);
		for (AbsCmdReq dCls : dClassesLv1) {
			inOut(dCls, ear, skin, eye);
		}
		fusion.setAlgQueue(noiron);
		return translateOut(fusion.act(ear, skin, eye));
	}

	public String getSoulEmotion() {
		return kokoro.getEmot();
	}

	public String getEmot() {
		// emot (emotion for display)
		String x1 = emot;
		switch (this.emot) {
		case "APCuss ":
			x1 = "angry";
			break;
		case "APDirtyTalk":
			x1 = "grinny";
			break;
		case "APMoan":
			x1 = "horny";
			break;
		case "APSay":
			x1 = "speaking";
			break;
		case "APSleep0":
			x1 = "dreaming";
			break;
		case "APSleep":
			x1 = "asleep";
			break;
		case "APSpell":
			x1 = "blank";
			break;
		default:
			break;
		}
		emot = "";
		return x1;
	}

	protected void inOut(AbsCmdReq dClass, String ear, String skin, String eye) {
		dClass.input(ear, skin, eye); // new
		dClass.output(noiron);
	}

	protected String translateIn(String earIn) {
		// makes sure the chobit doesn't feedback on her own output
		if (earIn.equals(lastOutput)) {
			return "";
		}
		return earIn;
	}

	protected String translateOut(String outResult) {
		// save last output served
		if (!outResult.isEmpty()) {
			lastOutput = outResult;
			this.timeGate.open();
			this.kokoro.standBy = false;
		}
		// standBy :
		else {
			if (this.timeGate.isClosed()) {
				this.kokoro.standBy = true;
				this.timeGate.open();
			} else {
				this.kokoro.standBy = false;
			}
		}
		return outResult;
	}

	@Override
	public String think(String ear, String skin, String eye) {
		return doIt(ear, skin, eye);
	}

	public Boolean getStandby() {
		return kokoro.standBy;
	}
}

File: livingrimoire start here/LivingGrimoire Kotlin/kotlin_core/ChobitsLight.kt
package com.yotamarker.lgkotlinfull.LGCore

import java.util.*


class ChobitsLight(personality: PersonalityLight) : thinkable {
    protected var emot:String = "" // emotion
        get() {
            // emot (emotion for display)
            var x1 = emot
            when (emot) {
                "APCuss " -> x1 = "angry"
                "APDirtyTalk" -> x1 = "grinny"
                "APMoan" -> x1 = "horny"
                "APSay" -> x1 = "speaking"
                "APSleep0" -> x1 = "dreaming"
                "APSleep" -> x1 = "asleep"
                "APSpell" -> x1 = "blank"
                else -> {
                }
            }
            emot = ""
            return x1
        }
    protected var dClassesLv1: ArrayList<AbsCmdReq>

    // algorithms fusion (polymarization)
    protected var AlgDurations: Hashtable<String, Int>
    protected var fusion: Fusion

    // region essential DClasses
    protected var permission: Permission? = null
    protected var dPermitter: DPermitter? = null

    // endregion
    protected var noiron: Neuron

    // sleep vars :
    protected var activePerson = Person()

    // added :
    protected var kokoro // soul
            : Kokoro
    protected var master = Person()
    protected var lastOutput = ""

    // standBy phase 260320
    protected var timeGate = ZeroTimeGate()
    fun loadPersonality(personality: PersonalityLight) {
        AlgDurations = personality.algDurations
        fusion = personality.fusion
        kokoro = personality.kokoro
        noiron = Neuron()
        dClassesLv1 = personality.getdClassesLv1()
    }

    fun doIt(ear: String, skin: String?, eye: String?): String {
        var ear = ear
        ear = translateIn(ear)
        for (dCls in dClassesLv1) {
            inOut(dCls, ear, skin, eye)
        }
        fusion.setAlgQueue(noiron)
        return translateOut(fusion.act(ear, skin, eye))
    }

    protected fun inOut(dClass: AbsCmdReq, ear: String?, skin: String?, eye: String?) {
        dClass.input(ear!!, skin!!, eye!!) // new
        dClass.output(noiron)
    }

    protected fun translateIn(earIn: String): String {
        // makes sure the chobit doesn't feedback on her own output
        return if (earIn == lastOutput) {
            ""
        } else earIn
    }

    protected fun translateOut(outResult: String): String {
        // save last output served
        if (!outResult.isEmpty()) {
            lastOutput = outResult
            timeGate.open()
            kokoro.standBy = false
        } else {
            if (timeGate.isClosed) {
                kokoro.standBy = true
                timeGate.open()
            } else {
                kokoro.standBy = false
            }
        }
        return outResult
    }

    override fun think(ear: String, skin: String, eye: String): String {
        return doIt(ear, skin, eye)
    }

    init {
        AlgDurations = personality.algDurations
        fusion = personality.fusion
        kokoro = personality.kokoro
        noiron = Neuron()
        dClassesLv1 = personality.getdClassesLv1()
    }
}